judoscale-ruby/lib/judoscale/config.rb

```diff
@@ -5,16 +5,43 @@
 
 module Judoscale
   class Config
-    class Dyno
-      attr_reader :name, :num
-
-      def initialize(dyno_string)
-        @name, @num = dyno_string.to_s.split(".")
-        @num = @num.to_i
+    class RuntimeContainer
+      # E.g.:
+      # :heroku, "worker_fast", "3"
+      # :render, "srv-cfa1es5a49987h4vcvfg", "5497f74465-m5wwr", "web" (or "worker", "pserv", "cron", "static")
+      def initialize(platform, service_name = nil, instance = nil, service_type = nil)
+        @platform = platform
+        @service_name = service_name
+        @instance = instance
+        @service_type = service_type
       end
 
       def to_s
-        "#{name}.#{num}"
+        # heroku: 'worker_fast.5'
+        # render: 'srv-cfa1es5a49987h4vcvfg.5497f74465-m5wwr'
+        "#{@service_name}.#{@instance}"
+      end
+
+      def web?
+        # NOTE: Heroku isolates 'web' as the required _name_ for its web process
+        # type, Render exposes the actual service type more explicitly
+        @service_name == "web" || @service_type == "web"
+      end
+
+      # Since Heroku exposes ordinal dyno 'numbers', we can tell if the current
+      # instance is redundant (and thus skip collecting some metrics sometimes)
+      # We don't have a means of determining that on Render though — so every
+      # instance must be considered non-redundant
+      def redundant_instance?
+        on_heroku? && @instance.to_i != 1
```

Comment:
This is the only place we use `on_heroku?` now, and `on_render?` isn't used at all. I think we should remove those two methods.

It also makes me wonder if we even need the `@platform` state in this object, since this is the only place we use it. We could instead check to see if `@instance` is numeric. wdyt?